### PR TITLE
Add CI workflow

### PR DESCRIPTION
**Pull request by embeddedt**
This is mostly for selfish reasons, so I don't have to recompile the mod locally every time you push a change. It also has the advantage of checking that PRs do actually compile.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        distribution: 'adopt'
+        java-version: '8'
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Build with Gradle
+      run: ./gradlew -Pnet.minecraftforge.gradle.disableUpdateChecker=true build
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v1
+      with:
+        name: more-refined-storage
```
Comment (tth05) [inline .github/workflows/build.yml]:
```suggestion
        name: morerefinedstorage.zip
```
Makes it a bit more clear what you get

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: '8'
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Build with Gradle
+      run: ./gradlew -Pnet.minecraftforge.gradle.disableUpdateChecker=true build
+    - name: Upload artifacts
```
Comment (tth05) [inline .github/workflows/build.yml:23]:
```suggestion
    - name: Rename jar
      run: mv build/libs/morerefinedstorage-*.jar build/libs/morerefinedstorage-${GITHUB_SHA::7}.jar
    - name: Upload artifacts
```
This avoids confusion when downloading multiple of these

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      uses: actions/upload-artifact@v1
```
Comment (tth05) [inline .github/workflows/build.yml]:
```suggestion
      uses: actions/upload-artifact@v2
```
Use `v2`, just to stay up-to-date :)

**Review by tth05 — CHANGES_REQUESTED** [review]
Cool idea!

File: .github/workflows/build.yml (hunk 1 above)
Comment (embeddedt) [reply .github/workflows/build.yml]: Won't this make the filename `morerefinedstorage.zip.zip`? I haven't tested it, so I'll try, but that would be ugly.

Comment (tth05) [reply .github/workflows/build.yml]: Oh yeah you're right. I only looked at the artifact name but not at the downloaded file name in my testing.

Comment (embeddedt) [reply .github/workflows/build.yml]: Okay, I reverted it. This should be good to go now.

**Review by tth05 — APPROVED** [review]

**Comment by tth05** [comment]
@embeddedt Just noticed it runs for each change in a PR as well, that's kind of annoying. Would you mind if I disable that?

**Comment by embeddedt** [comment]
@tth05 I mostly care about it for pushes to the main branch, so it's fine by me, but I'm curious what's annoying about it, as it shouldn't be sending any notifications unless it's failing.

**Comment by tth05** [comment]
> @tth05 I mostly care about it for pushes to the main branch, so it's fine by me, but I'm curious what's annoying about it, as it shouldn't be sending any notifications unless it's failing.

This is what I was thinking of:
- You get notifications for successful workflow runs 
- It wastes monthly github actions minutes
- It spams the actions tab with unnecessary workflow runs

**Comment by embeddedt** [comment]
> You get notifications for successful workflow runs

I didn't realize (or have forgotten) that that is a feature. I guess I have it turned off. :slightly_smiling_face: 

> It wastes monthly github actions minutes

Public repositories have unlimited minutes as far as I know, since they're considered open-source.

> It spams the actions tab with unnecessary workflow runs

Fair enough.

In any case, it doesn't bother me if you opt to disable that part.

**Pull request merged**